lib/Driver/ToolChains.cpp

```diff
@@ -1337,6 +1357,42 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   return {"clang++", Arguments};
 }
 
+std::string
+toolchains::Android::getTarget() const {
+  // Explicitly set the linker target to "androideabi", as opposed to the
+  // llvm::Triple representation of "armv7-none-linux-android".
+  // This is the only ABI we currently support for Android.
+  return "armv7-none-linux-androideabi";
+}
+
+bool toolchains::Android::shouldSpecifyTargetTripleToLinker() const {
+  return false;
+}
+
+bool toolchains::Android::shouldProvideRPathToLinker() const {
+  return false;
+}
+
+std::vector<std::string> toolchains::Android::getAdditionalLinkerPaths() const {
+  // "ANDROID_NDK_HOME" is set by the Swift build script. Still, emit a nice
+  // assertion message here for people building via CMake directly (even though
+  // this is not something we support).
+  const char *ndkhome = getenv("ANDROID_NDK_HOME");
+  assert(ndkhome && "Environment variable ANDROID_NDK_HOME needs to be set "
```

Comment:
But given the other discussion, we shouldn't be using an environment variable anyway.